### PR TITLE
Add ssh completer

### PR DESCRIPTION
**Pull request by jhdulaney**
Signed-off-by: John Dulaney jdulaney@fedoraproject.org

### CODE REVIEW

File: xonsh/completers/ssh.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+import os
+import subprocess
+
+def get_host_list():
+    '''
```
Comment (scopatz) [inline xonsh/completers/ssh.py]: PEP8 tripple double quotes, not triple single quotes.

File: xonsh/completers/ssh.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            else:
+                host_list.append(line)
+        return host_list
+
+
+def complete_ssh(prefix, line, begidx, endidx, ctx):
+    '''
+    Completes ssh hosts based on contents of ~/.ssh/known_hosts
+    '''
+    line_len = len(line.split())
```
Comment (scopatz) [inline xonsh/completers/ssh.py]: Where is `line_len` used?

**Comment by scopatz** [comment]
Thanks for putting this in @jhdulaney!  Just a couple of small issues.

**Comment by blahgeek** [comment]
SSH completion is already provided by bash-completion (after #1602 ), @jhdulaney would you mind give that a shot?

File: xonsh/completers/ssh.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            elif ':' in line:
+                host_list.append(line.split(':')[0].strip('[').strip(']'))
+            else:
+                host_list.append(line)
+        return host_list
+
+
+def complete_ssh(prefix, line, begidx, endidx, ctx):
+    '''
+    Completes ssh hosts based on contents of ~/.ssh/known_hosts
```
Comment (mitnk) [inline xonsh/completers/ssh.py]: Nice to see ssh completer coming!

Should the hosts be retrieved from `.ssh/config` file too?

**Comment by scopatz** [comment]
Hey @jhdulaney please let us know if there is anything you need to push this along.

**Comment by blarghmatey** [comment]
For what it's worth, this is what I put together in my `~/.xonshrc` file which is working well so far:
```python
def ssh_host_completer(prefix, line, begidx, endidx, ctx):
    if not line.startswith('ssh '):
        return None
    hostlist = []
    host_prefix = prefix.split('@')[-1]
    with open("{}/.ssh/known_hosts".format($HOME), 'r') as known_hosts:
        for hostline in known_hosts:
            namelist = hostline.split(' ')[0]
            hostlist += namelist.split(',')
    prefix_length = len(host_prefix)
    return (set([target for target in hostlist if host_prefix in target]), prefix_length)

completer add ssh ssh_host_completer
```

**Comment by gforsyth** [comment]
I'm closing this out. If anyone wants to work on it or see this reopened, just ping on the PR.

**Pull request closed without merging**